torch/storage.py

```diff
@@ -227,6 +237,40 @@ def untyped(self):
         return self
 
 
+def _share_memory_lock_protected(fn):
+    @functools.wraps(fn)
+    def wrapper(self, *args, **kwargs):
+        to_free = None
+        to_wait = None
+        with _share_memory_lock:
+            key = self._cdata
+            if key in _share_memory_map:
+                to_wait = _share_memory_map[key]
+            else:
+                _share_memory_map[key] = threading.RLock()
+                _share_memory_map[key].acquire()
```

Comment:
Checking my understanding - is this map supposed to be key'd by tensor storage?

`self._cdata` prints the TensorImpl pointer, which will be different for tensor aliases (tensors sharing an underlying storage but that have their own TensorImpls).
```
>>> a = torch.ones(2)
>>> b = a.view(-1)
>>> a._cdata
101027888
>>> b._cdata
101564736
```

Reply:
These are Storage methods. So `self._cdata` will point to the StorageImpl here. So that will be shared as expected.
Multiple Storage pointing to the same data_ptr is a nono so no need to worry about it.

Reply:
Ah whoops - thanks!